src/codegen/function_body_chunk.rs

```diff
@@ -119,15 +172,289 @@ impl Builder {
+                        format!("let {0}_data: Box_<Box_<Option<{1}>>> = Box::new(Box::new({0}.into()));",
+                                trampoline.name, trampoline.bound_name)));
+            }
+        } else if !is_destroy {
+            if trampoline.scope.is_call() {
+                chunks.push(Chunk::Custom(format!("let {0}_data: &{1} = &{0};",
+                                                  trampoline.name,
+                                                  trampoline.bound_name)));
+            } else {
+                chunks.push(Chunk::Custom(format!("let {0}_data: Option<{1}> = {0}.into();",
+                                                  trampoline.name,
+                                                  trampoline.bound_name)));
+            }
+        }
+
+        let mut body = Vec::new();
+        let mut arguments = Vec::new();
+
+        for par in trampoline.parameters.transformations.iter() {
+            if trampoline.parameters.c_parameters[par.ind_c].c_type == "gpointer" ||
+               par.name == "this" {
+                continue;
+            }
+            let type_ = env.type_(par.typ);
+            match *type_ {
+                library::Type::Fundamental(ref x) if !x.requires_conversion() => {}
+                library::Type::Fundamental(library::Fundamental::Boolean) |
+                library::Type::Fundamental(library::Fundamental::UniChar) => {
+                    body.push(Chunk::Custom(format!("let {0} = from_glib({0});", par.name)));
+                }
+                _ => {
+                    let (begin, end) = ::codegen::translate_from_glib::from_glib_xxx(par.transfer, None);
+                    body.push(Chunk::Custom(format!("let {1} = {0}{1}{2};", begin, par.name, end)));
+                }
+            }
+            arguments.push(Chunk::Name(par.name.clone()));
+        }
+
+        let func = trampoline.parameters
+                             .c_parameters
+                             .last()
+                             .map(|p| p.name.clone())
+                             .unwrap_or_else(|| "Unknown".to_owned());
+
+        if let Some(ref full_type) = full_type {
+            body.push(
+                Chunk::Let {
+                    name: format!("{}callback", if is_destroy { "_" } else { "" }),
+                    is_mut: false,
+                    value: Box::new(Chunk::Custom(format!("Box_::from_raw({} as *mut _)", func))),
+                    type_: Some(Box::new(Chunk::Custom(full_type.clone()))),
+                }
+            );
+            if !is_destroy {
+                if trampoline.scope.is_async() {
+                    body.push(
+                        Chunk::Custom(
+                            format!("{}let callback = callback.{}.expect(\"cannot get closure...\");",
+                                    if trampoline.ret.c_type != "void" { "let res = " } else { "" },
+                                    pos)));
```

Comment:
This can cause  `let res = let callback = callback.0.expect)`, not sure that it planned.
Same block on line 349.

Reply:
No it's not, good catch!